src/main/java/ui/UI.java

```diff
@@ -540,4 +544,27 @@ public boolean isWindowMinimized() {
     public boolean isWindowFocused() {
         return mainStage.isFocused();
     }
+
+    /**
+     * Warns user if the Java runtime version is lower than HT's requirement
+     */
+    private void warnIfJavaVersionOutdated() {
+        try {
+            JavaVersion runtimeVersion = JavaVersion.fromString(System.getProperty("java.runtime.version"));
+            JavaVersion requiredVersion = JavaVersion.fromString(REQUIRED_VERSION);
+
+            if (JavaVersion.isJavaVersionTooLow(runtimeVersion, requiredVersion)) {
+                showJavaVersionOutdatedWarning(runtimeVersion, requiredVersion);
+            }
+        } catch (IllegalArgumentException e) {
+            logger.error("Java Version string failed to be parsed.", e);
+        }
+    }
+
+    private void showJavaVersionOutdatedWarning(JavaVersion runtimeVersion, JavaVersion requiredVersion) {
+        String message = String.format(JavaVersion.OUTDATED_JAVA_VERSION_MESSAGE,
+                requiredVersion.toString(), runtimeVersion.toString());
```

Comment:
8 space indentation is a last resort. See if there is a more readable choice. e.g. 
```java
String message = String.format(JavaVersion.OUTDATED_JAVA_VERSION_MESSAGE,
                               requiredVersion.toString(), runtimeVersion.toString());
```